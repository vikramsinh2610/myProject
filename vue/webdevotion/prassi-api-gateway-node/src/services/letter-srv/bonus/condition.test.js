const Condition = require('./condition');

describe('Condition', () => {
  test('should have these properties', () => {
    const condition = new Condition({ targets: [{}] });
    expect(condition).toHaveProperty('targets');
  });

  test('should return it s JSON Schema', () => {
    expect(Condition.getJSONSchema()).toMatchSnapshot();
  });
});
