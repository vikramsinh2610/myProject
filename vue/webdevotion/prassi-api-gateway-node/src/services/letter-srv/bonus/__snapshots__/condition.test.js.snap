// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condition should return it s JSON Schema 1`] = `
Object {
  "properties": Object {
    "targets": Object {
      "items": Object {
        "properties": Object {
          "kpi": Object {
            "properties": Object {
              "_id": Object {
                "type": "string",
              },
              "options": Object {
                "additionalProperties": true,
                "type": "object",
              },
              "type": Object {
                "enum": Array [
                  "number",
                  "currency",
                  "percentage",
                ],
                "type": "string",
              },
              "value": Object {
                "type": "integer",
              },
            },
            "required": Array [
              "_id",
            ],
            "type": "object",
          },
          "targetValue": Object {
            "type": "integer",
          },
          "weightPercentage": Object {
            "maximum": 10000,
            "minimum": 0,
            "type": "integer",
          },
        },
        "required": Array [
          "kpi",
          "targetValue",
          "weightPercentage",
        ],
        "type": "object",
      },
      "type": "array",
    },
  },
  "required": Array [
    "targets",
  ],
  "type": "object",
}
`;
